beacon-chain/powchain/log_processing.go

```
@@ -301,6 +312,15 @@ func (s *Service) processPastLogs(ctx context.Context) error {
 		}
 		logs, err := s.httpLogger.FilterLogs(ctx, query)
 		if err != nil {
+			if tooMuchDataRequestedError(err) {
+				if batchSize == 0 {
+					return err
```

maybe return a different error here
ex:
```
batch size is now zero
```